### PR TITLE
Add flag to drop quality of low-pt TkMuons without an SA match

### PR DESCRIPTION
**Pull request by epalencia**
This PR includes a flag to duplicate the tkMuonGMT collection which degrades the quality flag of those muons without an associated SA muon. This is a temporary fix until we re-optimize the ID to further control rate.

The corresponding PR in the cms-l1t-offline IB is https://github.com/cms-l1t-offline/cmssw/pull/1109

### CODE REVIEW

**Comment by epalencia** [comment]
Please test

**Comment by epalencia** [comment]
Please test

**Comment by epalencia** [comment]
@perrotta, tests have been running for too long, is there any known issue? Should they be restarted?

**Comment by aloeliger** [comment]
@epalencia I think Andrea is on vacation right now? It looks like there was some sort of server error on the job (``` urllib2.HTTPError: HTTP Error 500: Internal Server Error' -gt 0 ']'```) after it finished that prevented buildbot from reporting back here or from timing out. You may have to forcibly abort the tests and restart them. Perhaps @rappoccio has a better solution though?

**Comment by epalencia** [comment]
please abort

**Comment by epalencia** [comment]
Please test

File: L1Trigger/Phase2L1GMT/plugins/Phase2L1TGMTFilter.cc [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  iEvent.getByToken(srcMuons_, muonHandle);
+
+  std::vector<l1t::TrackerMuon> out;
+
+  for (uint i = 0; i < muonHandle->size(); ++i) {
+    auto mu = muonHandle->at(i);
+    bool noSAMatch = true;
+    if (applyLowPtFilter_) {
+      if ((fabs(mu.phEta()) < 0.9 && mu.phPt() < ptBarrelMin_) ||
+          (fabs(mu.phEta()) > 0.9 && mu.phPt() < ptEndcapMin_)) {
```
Comment (rappoccio) [inline L1Trigger/Phase2L1GMT/plugins/Phase2L1TGMTFilter.cc:67]: This 0.9 should be a configurable parameter.

**Review by rappoccio — CHANGES_REQUESTED** [review]

File: L1Trigger/Phase2L1GMT/plugins/Phase2L1TGMTFilter.cc (hunk 1 above)
Comment (folguera) [reply L1Trigger/Phase2L1GMT/plugins/Phase2L1TGMTFilter.cc:67]: @rappoccio Do you want this to be in the cfg file? Or should we define it somewhere in the code? the limit between barrel and endcap is a fixed value.  Thanks.

Comment (epalencia) [reply L1Trigger/Phase2L1GMT/plugins/Phase2L1TGMTFilter.cc]: @folguera provided the requested update.

**Comment by epalencia** [comment]
Please test

**Pull request merged**